### ISSUE TITLE
libertempo 

### ISSUE THREAD

**Issue by olivierb987** [wouldsmina/Libertempo#958]
bonjour , 

je ne parviens pas a faire fonctionner l'authentification CAS sur une version 1.13

les fichiers de configuration config_CAS.php est correctement renseigné et référence le serveur ainsi que le port et l'URI de mon serveur CAS . 
le mode de connexion (how_to_connect_user)  est bien valué a "cas"
l'authentification LDAP fonctionne quant a elle correctement .

voici ce que j'obtiens lors de mes tentatives de connexion apres avoir activé le mode dev dans le fichier env.php
Notice: Undefined index: config in /var/www/html/Libertempo/includes/fonction.php on line 270

Notice: Undefined index: config in /var/www/html/Libertempo/includes/fonction.php on line 271

Notice: Undefined index: config in /var/www/html/Libertempo/includes/fonction.php on line 272

Notice: Undefined index: config in /var/www/html/Libertempo/includes/fonction.php on line 273

Error: Internal script failure

les lignes de code correspondantes dans le fichier functions.php sont les suivantes 

function authentification_passwd_conges_CAS()
{
    $config_CAS_host       =$_SESSION['config']['CAS_host'];
    $config_CAS_portNumber =$_SESSION['config']['CAS_portNumber'];
    $config_CAS_URI        =$_SESSION['config']['CAS_URI'];
    $config_CAS_CACERT     =$_SESSION['config']['CAS_CACERT'];
    $config = new \App\Libraries\Configuration(\includes\SQL::singleton());

j'ai évidement tenté de forcer les variables directement dans le fichier fonctions.php , mais sans effets 

savez vous comment faire fonctionner CAS ? 

merci pour vos remarques !

**Comment by wouldsmina** [comment]
Bonjour,

Malheureusement l'authentification cas ne fonctionne pas sur cette version. Il est peut-être possible de passer par le module apache.

**Comment by olivierb987** [comment]
Bonjour et merci pour cette réponse rapide , 
en avez vous un exemple ?

**Comment by wouldsmina** [comment]
Pas pour libertempo particulièrement, mais voici un tuto qui pourrait t'aider : https://iam.uconn.edu/mod_auth_cas-installation-and-configuration/

Tiens moi au courant ;)

**Comment by olivierb987** [comment]
merci pour le lien , je vais chercher dans cette direction

**Comment by olivierb987** [comment]
Avec l'aide de ton doc, j'ai bien installé et configuré le module d'authentification CAS / Apache. Je me suis inspiré également de ce post (https://github.com/libertempo/web/pull/814)  et en particulier de ce qu'a posté l'utilisateur @ngombe42
Pour une raison que j'ignore, j'ai du valuer à la main les variables de la fonction  $config_CAS_host , $config_CAS_portNumber $config_CAS_URI et $config_CAS_CACERT de la fonction  authentification_passwd_conges_CAS() .
Actuellement , je vois bien la validation du ticket coté CAS , mais il est toujours impossible d'entrer dans l'application.
Quelques lignes de debug plus tard dans la fonction d'authentification , j'ai l'impression que c'est la variable $userPT qui n'est pas renseignée tandis que je récupère bien le $usernameCAS de l'utilisateur qui se connecte .
si vous avez des idées …

**Comment by wouldsmina** [comment]
Salut, 
Je crois qu'il faut configurer libertempo pour s'authentifier en local puisque c'est apache qui s'occupe de la partie cas.

**Comment by wouldsmina** [comment]
Si tu es à l'aise avec php et le fonctionnement de CAS, je peux te donner les dernières lignes de code que j'ai fait à ce sujet. J'avais réussi à transmettre le PT au serveur CAS depuis l'api, mais mon CAS refuse de le valider pour une raison qui m'échappe.

**Comment by olivierb987** [comment]
salut , avec plaisir ! Pour le moment, je ne parviens toujours pas a faire valider mon utilisateur authentifié par CAS et je boucle sur le message d'erreur "utilisateur inconnu" de libertempo

**Comment by wouldsmina** [comment]
Juste au cas où... Lorsque tu as créé le premier utilisateur durant l'installation, tu as bien utilisé un identifiant qui existe dans ton ldap ? C'est avec cet utilisateur que tu as cet erreur ?

**Comment by olivierb987** [comment]
Oui, je te confirme. c'est bien un utilisateur du ldap et qu'on trouve également dans la db

**Comment by wouldsmina** [comment]
OK, je vais essayer de trouver un peu de temps pour te donner le code et les infos qui pourraient te permettre d'avancer. En attendant, tu peux déjà jeter un œil sur le [pull request](https://github.com/libertempo/api/pull/141 ) sur l'authentification cas dans le projet API.
